java/code/src/com/redhat/rhn/domain/product/SUSEProductFactory.java

```diff
@@ -105,6 +105,14 @@ public static List<SUSEProductSCCRepository> allProductRepos() {
         return c.list();
     }
 
+    /**
+     * @return return true if any products are available, otherwise false
+     */
+    public static boolean hasProducts() {
+        return getSession().createQuery("SELECT productId FROM SUSEProduct")
+                .stream().findAny().isPresent();
```

Comment:
I think we can avoid extracting all the products. Hibernate should be using a scrollable result set when using `stream()`, but still we can avoid a select all from the table by doing directly the check in the query:

```suggestion
        return getSession().createQuery("SELECT count(p) > 0 FROM SUSEProduct p", Boolean.class)
            .uniqueResult();
```